src/Utilities/Utilities.jl

```diff
@@ -3,4 +3,18 @@ module Utilities
 include("plushalf.jl")
 include("unrolled_functions.jl")
 
+"""
+    Utilities.remove_from_cache!(object)
+
+Remove `object` from the cache of created objects (only topologies and
+grids are cached). In most cases, this should not need to be called, unless you
+are e.g. doing a sweep over grid paramaters.
+
+Only call this if you are completely finished with this object configuration, and
+no other instances of it exist.
+"""
+function remove_from_cache!(object) # fallback
+end
```

Comment:
I don't want to overengineer this, but maybe it would be useful to also have a `clean_caches!` function that cleans all the caches. I am thinking that this might be useful when doing interactive development to ensure that we are not using stale objects. 

Can you please also describe this caching mechanism somewhere in the main docs?